docs/test_docs.py

```diff
@@ -25,13 +27,11 @@ def test_rest_api_version():
     assert jupyterhub_version == rest_api_version
 
 
-def test_restapi_scopes():
+def test_rest_api_rbac_scope_descriptions_are_updated():
+    """Checks that the RBAC scope descriptions in JupyterHub's REST API
+    definition file (rest-api.yml) as can be updated by generate-scope-table.py
+    matches what is committed."""
     run([sys.executable, "source/rbac/generate-scope-table.py"], cwd=here, check=True)
-    run(
-        ['pre-commit', 'run', 'prettier', '--files', 'source/_static/rest-api.yml'],
-        cwd=here,
-        check=False,
-    )
```

Comment:
This is version controlled and we have pre-commit run in other tests so no need to do it here as well.

Reply:
This pre-commit step is required for the test to pass. `generate-scope-table` above will generate a file with correct contents, but _not_ correct formatting.

The failure is being hidden because there's a bug a few lines below, where `--no-pager` must be passed _before_ `diff`:

```python
"git", "--no-pager", "diff", "--exit-code"...
```

(don't ask me why passing invalid options doesn't set an error code!)

Reply:
Ah hmmmm, since we have a dedicated pre-commit test, I'd argue that we let this test not run pre-commit also. What do you think?

- This test would check that the rest-api is updated
- The pre-commit test would check that all (updated) files are pre-commit verified

Reply:
Oh nevermind I get it now! I kind of got it, then confused myself to think we didn't need pre-commit again, and now I'm back to thinking we need it.

Then: do we want it as part of the generate script or as part of the test?

Reply:
We can also update the test to do a diff of the parsed content. Then formatting shouldn't matter. However, then _presentation_ of the diff is quite a bit harder. That shouldn't matter too much, as it's an auto-generated file.

Reply:
I ended up making the pre-commit run as part of running the generate-scope-table.py script, I had to add pre-commit as a docs/requirements.txt for this. Acceptable?

Reply:
Sure, I think that's fine. That'll prevent the pre-commit error on commit after run, too. Main disadvantage I see is that it might slow down iterating on the docs, since running pre-commit becomes part of the rebuilding the docs cycle. But maybe it won't run if the file hasn't changed.

We can switch to the structural comparison in the test if it turns out to be a problem.